Monitors/remote_mount.py

```diff
@@ -0,0 +1,140 @@
+    elif unit == 'MB':
+        _size_bytes = value * 1024 * 1024
+    elif unit == 'KB':
+        _size_bytes = value * 1024
+    elif unit in ['BYTES', 'BYTE', 'B', '']:
+        _size_bytes = value
+
+    return _size_bytes
+
+
+@register
+class RemoteMountMonitor(RemoteMonitor):
+    type = "remotemount"
+
+    def __init__(self, name, config_options):
+        RemoteMonitor.__init__(self, name, config_options)
+
+        self._free_space = RemoteMonitor.get_config_option(config_options, 'free_space', required=False, default='1GB')
+
+        # free space can be given as a percentage or an absolute size (i.e. 10GB)
+        if '%' in self._free_space:
+            self.free_space_required = int(self._free_space.replace('%', ''))
+            self.free_space_unit = 'percent'
+            if self.free_space_required > 100 or self.free_space_required < 0:
+                raise MonitorConfigurationError("invalid free space percentage")
+        else:
+            self.free_space_required = _size_string_to_bytes(self._free_space)
+            self.free_space_unit = 'byte'
+            if self.free_space_required < 0:
+                raise MonitorConfigurationError("invalid negative free space")
+
+    def run_test(self):
+        mounts = self.get_mounts()
+
+        failed_mounts = []
+
+        if self.free_space_unit not in ['percent', 'byte']:
+            raise AssertionError('the unit of free space must be one of: ["percent", "byte"]')
+
+        # Father all the mounts that do not have enough free space
+        if self.free_space_unit == 'percent':
+            for mount in mounts:
+                percent_free = 100 - mount.get('Use%')
+                if percent_free < self.free_space_required:
+                    failed_mounts.append(mount)
+        elif self.free_space_unit == 'byte':
+            for mount in mounts:
+                free_bytes = mount.get('Avail') - mount.get('Used')
+                if free_bytes < self.free_space_required:
+                    failed_mounts.append(mount)
+
+        if not failed_mounts:
+            return self.record_success()
+
+        msg = ''
+        for mount in failed_mounts:
+            if msg != '':
+                msg += '\n'
+            mount_point = mount.get("Mounted on")
+            percent_left = '{}%'.format(100 - mount.get("Use%"))
+            mount_size = _bytes_to_size_string(mount.get("Avail"))
+            amount_left = _bytes_to_size_string(mount.get('Avail') - mount.get('Used'))
+            msg += '{}:{} has {} free space out of {} ({} left)'.format(self.host, mount_point, percent_left,
+                                                                        mount_size, amount_left)
+        return self.record_fail(msg)
+
+    def describe(self):
+        pass
+
+    def get_params(self):
+        return super(RemoteMountMonitor, self).get_params() + (self._free_space,)
+
+    def get_mounts(self):
+        result = self.connection.run('df --output', hide=True)

```

Comment:
`--output` looks to be Linux only? I see the docs mention this monitor is for Linux hosts; how does it handle the error of being run on a host which doesn't support it?